hooks/repo.go

```diff
@@ -0,0 +1,144 @@
+package hooks
+
+import (
+	"fmt"
+
+	"github.com/prebid/prebid-server/hooks/hookstage"
+)
+
+type HookRepository interface {
+	GetEntrypointHook(id string) (hookstage.Entrypoint, bool)
+	GetRawAuctionHook(id string) (hookstage.RawAuction, bool)
+	GetProcessedAuctionHook(id string) (hookstage.ProcessedAuction, bool)
+	GetBidRequestHook(id string) (hookstage.BidRequest, bool)
+	GetRawBidResponseHook(id string) (hookstage.RawBidResponse, bool)
+	GetAllProcessedBidResponsesHook(id string) (hookstage.AllProcessedBidResponses, bool)
+	GetAuctionResponseHook(id string) (hookstage.AuctionResponse, bool)
+}
+
+func NewHookRepository(hooks map[string]interface{}) (HookRepository, error) {
+	repo := new(hookRepository)
+	for id, hook := range hooks {
+		if err := repo.add(id, hook); err != nil {
+			return nil, err
+		}
+	}
+
+	return repo, nil
+}
+
+type hookRepository struct {
+	entrypointHooks              map[string]hookstage.Entrypoint
+	rawAuctionHooks              map[string]hookstage.RawAuction
+	processedAuctionHooks        map[string]hookstage.ProcessedAuction
+	bidRequestHooks              map[string]hookstage.BidRequest
+	rawBidResponseHooks          map[string]hookstage.RawBidResponse
+	allProcessedBidResponseHooks map[string]hookstage.AllProcessedBidResponses
+	auctionResponseHooks         map[string]hookstage.AuctionResponse
+}
+
+func (r *hookRepository) GetEntrypointHook(id string) (h hookstage.Entrypoint, ok bool) {
+	return getHook(r.entrypointHooks, id)
+}
+
+func (r *hookRepository) GetRawAuctionHook(id string) (hookstage.RawAuction, bool) {
+	return getHook(r.rawAuctionHooks, id)
+}
+
+func (r *hookRepository) GetProcessedAuctionHook(id string) (hookstage.ProcessedAuction, bool) {
+	return getHook(r.processedAuctionHooks, id)
+}
+
+func (r *hookRepository) GetBidRequestHook(id string) (hookstage.BidRequest, bool) {
+	return getHook(r.bidRequestHooks, id)
+}
+
+func (r *hookRepository) GetRawBidResponseHook(id string) (hookstage.RawBidResponse, bool) {
+	return getHook(r.rawBidResponseHooks, id)
+}
+
+func (r *hookRepository) GetAllProcessedBidResponsesHook(id string) (hookstage.AllProcessedBidResponses, bool) {
+	return getHook(r.allProcessedBidResponseHooks, id)
+}
+
+func (r *hookRepository) GetAuctionResponseHook(id string) (hookstage.AuctionResponse, bool) {
+	return getHook(r.auctionResponseHooks, id)
+}
+
+func (r *hookRepository) add(id string, hook interface{}) (err error) {
+	var isCompatible bool
+
+	if h, ok := hook.(hookstage.Entrypoint); ok {
+		isCompatible = true
+		if r.entrypointHooks, err = addHook(r.entrypointHooks, h, id); err != nil {
+			return
```

Comment:
The empty return statements had me do a double take. This is valid Go, but IMHO it's easier to read if we `return err` on these lines instead of relying on a named return.